new-packages/ts-project/__tests__/state.test.ts

```diff
@@ -0,0 +1,695 @@
+              "jsDoc": undefined,
+              "kind": 303,
+              "localSymbol": undefined,
+              "modifierFlagsCache": 0,
+              "modifiers": undefined,
+              "name": IdentifierObject {
+                "end": 62,
+                "escapedText": "type",
+                "flags": 0,
+                "flowNode": undefined,
+                "jsDoc": undefined,
+                "kind": 80,
+                "modifierFlagsCache": 0,
+                "parent": undefined,
+                "pos": 56,
+                "symbol": undefined,
+                "transformFlags": 0,
+              },
+              "parent": undefined,
+              "pos": 56,
+              "questionToken": undefined,
+              "symbol": undefined,
+              "transformFlags": 0,
+            },
+            "type": "state_property_unhandled",
+          },
+        ],
+      ],
+    ]
+  `);
+});
+
+test('Should extract state.history with value "shallow"', async () => {
+  const tmpPath = await testdir({
+    'tsconfig.json': JSON.stringify({}),
+    'index.ts': ts`
+      import { createMachine } from "xstate";
+
+      createMachine({
+        history: "shallow",
+        states: {},
+      });
+    `,
+  });
+
+  const project = await createTestProject(tmpPath);
+
+  expect(project.extractMachines('index.ts')).toMatchInlineSnapshot(`
+    [
+      [
+        {
+          "edges": [],
+          "rootState": {
+            "data": {
+              "description": undefined,
+              "entry": [],
+              "exit": [],
+              "history": "shallow",
+              "initial": undefined,
+              "invoke": [],
+              "metaEntries": [],
+              "tags": [],
+              "type": undefined,
+            },
+            "id": "",
+            "states": [],
+          },
+          "states": [],
+        },
+        [],
+      ],
+    ]
+  `);
+});
+test('Should extract state.history with value "deep"', async () => {
+  const tmpPath = await testdir({
+    'tsconfig.json': JSON.stringify({}),
+    'index.ts': ts`
+      import { createMachine } from "xstate";
+
+      createMachine({
+        history: "deep",
+        states: {},
+      });
+    `,
+  });
+
+  const project = await createTestProject(tmpPath);
+
+  expect(project.extractMachines('index.ts')).toMatchInlineSnapshot(`
+    [
+      [
+        {
+          "edges": [],
+          "rootState": {
+            "data": {
+              "description": undefined,
+              "entry": [],
+              "exit": [],
+              "history": "deep",
+              "initial": undefined,
+              "invoke": [],
+              "metaEntries": [],
+              "tags": [],
+              "type": undefined,
+            },
+            "id": "",
+            "states": [],
+          },
+          "states": [],
+        },
+        [],
+      ],
+    ]
+  `);
+});
+
+test('Should extract state.history with value undefined', async () => {
```

Comment:
this one actually has some meaning for us, it's likely a redundant version of `history: 'shallow'`